pkg/logger/logger.go

```diff
@@ -65,6 +66,16 @@ func NewWith(cfgFn func(*zap.Config)) (Logger, error) {
 	return &logger{core.Sugar()}, nil
 }
 
+// NewWasm returns a new Logger to be used in a Wasm binary that will proxy the logs to the host.
+func NewWasm(log func(respptr unsafe.Pointer, respptrlen int32)) Logger {
```

Comment:
Is it possible to move the `log` func definition as well? Since we don't need to swap it out, and it looks pretty foreign.
```go
//go:wasmimport env log
func log(respptr unsafe.Pointer, respptrlen int32)
```

Reply:
@jmank88 Do you mean move the goimport into this package? I'm not 100% keen on this since it could potentially make it harder to track what imports are being used. Atm they are all listed in runner_wasip1 so our imports (not the WASI ones) are at least clearly defined.

Reply:
I think this needs to live in the runner_wasip1.go in order for the wasmimport to work. I even get a static check error when moving it to the logger. Can you confirm @cedric-cordenier?

Reply:
I guess I'm not following how these pieces fit together. The file that contains that func does not contain `//go:build wasip1` like many other files, so what is the difference with putting inside of this package?

If we are stuck for some reason, then maybe a better alternative is something like:
```suggestion
func NewWithSync(io.Writer) Logger {
```
With the caller passing in `wasm.WriteSyncer` :shrug:

Reply:
I think it has to do with the name of the file, its called: `runner_wasip1.go` if you modify the name you get a `missing function body` error. thats why @cedric-cordenier  mentioned about creating `logger_wasip1.go`. I can def modify the Constructor to receive a io.Writer